packages/core/src/root.ts

```diff
@@ -611,12 +611,11 @@ function tryHotSwappingValues(
     return [IGNORE];
   }
 
-  seen.add(currentValue);
```

Comment:
Same here, now **only** types/values that we have special handling for hot-swapping will be marked as seen.